### PR TITLE
Check scaleset size for missing nodes

### PR DESCRIPTION
**Pull request by bmc-msft**

### CODE REVIEW

File: src/api-service/__app__/onefuzzlib/workers/scalesets.py [hunk 1]
```diff
@@ -449,6 +449,29 @@ def _resize_shrink(self, to_remove: int) -> None:
         for node in nodes:
             node.send_stop_if_free()
 
+    def check_scaleset_size(self) -> None:
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/workers/scalesets.py:452]: Can we add a doc comment summarizing the responsibilities of this function?

Also, I find the name a bit confusing (`check` makes me think of a read-only predicate or guard function, but this does a lot of mutating). How about something like `sync_scaleset_size()`?

File: src/api-service/__app__/onefuzzlib/workers/scalesets.py [hunk 2]
```diff
@@ -449,6 +449,29 @@ def _resize_shrink(self, to_remove: int) -> None:
...
+        if self.state != ScalesetState.running:
+            return
+
+        size = get_vmss_size(self.scaleset_id)
+        if size is None:
+            logging.info(
+                SCALESET_LOG_PREFIX + "scaleset is unavailable. scaleset_id:%s",
+                self.scaleset_id,
+            )
+            self.set_shutdown(now=True)
```
Comment (ranweiler) [inline src/api-service/__app__/onefuzzlib/workers/scalesets.py:462]: If the scaleset is unavailable, why does it make sense to `set_shutdown()`?

**Review by ranweiler — APPROVED** [review]

File: src/api-service/__app__/onefuzzlib/workers/scalesets.py (hunk 2 above)
Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/workers/scalesets.py:462]: The scaleset shutdown code handles cleaning up all of the resources for the scaleset.  This includes deleting nodes, stopping tasks on said nodes, deleting scaleset specific storage queues, etc.

Comment (ranweiler) [reply src/api-service/__app__/onefuzzlib/workers/scalesets.py:464]: So is this just an optimization? If we don't call this here, what happens?

Comment (bmc-msft) [reply src/api-service/__app__/onefuzzlib/workers/scalesets.py:467]: The "Azure says this scaleset doesn't exist" error handling is to move to shutting it down.  I copied the comment from the other places we handle this same error condition.

**Pull request merged**